### PR TITLE
Extract ResourcesDef, MutatorsDef, OwnersDef interfaces

### PR DESCRIPTION
**Pull request by lannka**
for #23311

### CODE REVIEW

File: src/service/resources-impl.js [hunk 1]
```diff
@@ -70,6 +70,315 @@ let MarginChangeDef;
...
+   */
+  ampInitComplete() {}
+
+  /**
+   * Requires the layout of the specified element or top-level sub-elements
+   * within.
+   * @param {!Element} element
+   * @param {number=} opt_parentPriority
+   * @return {!Promise}
+   * @restricted
```
Comment (lannka) [inline src/service/resources-impl.js]: @jridgewell this `restricted` annotation is copied over. but do you think it's needed here? This method should be public, it's used outside of this file.

File: src/service/resources-impl.js [hunk 2]
```diff
@@ -70,6 +70,315 @@ let MarginChangeDef;
...
+}
+
+/**
+ * @interface
+ */
+export class ResourcesDef extends MutatorsAndOwnersDef {
+  /**
+   * Returns a list of resources.
+   * @return {!Array<!Resource>}
+   * @export
```
Comment (lannka) [inline src/service/resources-impl.js]: @jridgewell could you explain what does `@export` do here?

File: src/service/resources-impl.js [hunk 3]
```diff
@@ -70,6 +70,315 @@ let MarginChangeDef;
...
+  collapseElement(element) {}
+
+  /**
+   * Runs the specified measure, which is called in the "measure" vsync phase.
+   * This is simply a proxy to the privileged vsync service.
+   *
+   * @param {function()} measurer
+   * @return {!Promise}
+   */
+  measureElement(measurer) {}
```
Comment (jridgewell) [inline src/service/resources-impl.js]: I don't think these belong as an Owners interface, they're meant to be used by elements themselves.

File: src/service/resources-impl.js (hunk 2 above)
Comment (jridgewell) [reply src/service/resources-impl.js]: https://github.com/google/closure-compiler/wiki/Annotating-JavaScript-for-the-Closure-Compiler#export-export-sometype

File: src/service/resources-impl.js (hunk 1 above)
Comment (jridgewell) [reply src/service/resources-impl.js]: No, I don't think it's needed.

**Comment by lannka** [comment]
> Why are we splitting up the interfaces like this, where one extends another? It seems like multiple interfaces that the Resources class would then implement would be better for inabox?

Yep, that's the end goal. This is an easy first step. Will split those to separate services in following up PR.

File: src/service/resources-impl.js (hunk 3 above)
Comment (lannka) [reply src/service/resources-impl.js]: This is not part of Owners interface, it's part of Mutators interface

File: src/service/resources-impl.js (hunk 1 above)
Comment (lannka) [reply src/service/resources-impl.js]: removed

**Pull request merged**